app/Modules/Availability/AvailabilityService.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+namespace App\Modules\Availability;
+
+use App\Exceptions\OverlappingAvailabilityException;
+use App\Exceptions\PrivilegeException;
+use App\Modules\Availability\AvailabilityInPastException;
+use App\User;
+use Carbon\Carbon;
+
+class AvailabilityService
+{
+    /** @var mixed App\User */
+    protected $user;
+
+    /** HIS MIN TIME BLOCK TIME AVAILABILITY SHOULD BE */
+    const MIN_BLOCK_TIME = 30;
+
+    public function __construct(User $user)
+    {
+        $this->user = $user;
+    }
+
+    /**
+     * Validate that two given times are valid within the context of other availability.
+     *
+     * @param  Carbon  $from
+     * @param  Carbon  $to
+     * @param int $doNotCheck - Pass in an int to exclude from the search
+     * @param  User  $user - User
+     * @param  int|null  $excluded  - Availability ID to be excluded from the check.
+     * @throws AvailabilityInPastException - Cannot add availability that occurs in the past
+     * @return bool
+     */
+    public function validateAvailabilityTimes(Carbon $from, Carbon $to, User $user, int $doNotCheck =  NULL): bool
+    {
+        if ($from->isAfter($to)) {
+            return false;
+        }
+
+        if ($from->isBefore(new Carbon())) {
+            throw new AvailabilityInPastException();
+        }
+
+        if ($from->diffInMinutes($to) < self::MIN_BLOCK_TIME) {
+            throw new AvailabilityMinimumTimeException(self::MIN_BLOCK_TIME);
+        }
+
+        $avail = Availability::where([['user_id', $user->id], ['id', '!=', $doNotCheck]]);
+
+        // Find between the times being booked for
+        $avail->where(function ($query) use (&$from, &$to) {
+            // Where the start date is inside the booked time
+            $query->where(function ($query) use (&$from, &$to) {
+                $query->where('from', '>', $from)
+                    ->where('from', '<', $to);
+                // Or where the end date is inside the booked time
+            })->orWhere(function ($query) use (&$from, &$to) {
+                $query->where('to', '>', $from)
+                    ->where('to', '<', $to);
+                // Or where the times are the same
+            })->orWhere(['from' => $from, 'to' => $to]);
+        });
+
+        return $avail->doesntExist();
+    }
+
+    public function createAvailability(array $availabilityData): Availability
+    {
+        ['from' => $from, 'to' => $to, 'user_id' => $userId] = $availabilityData;
+
+        $availabilityUser = $this->user::findOrFail($userId);
+
+        try {
+            $passesTimeChecks = $this->validateAvailabilityTimes($from, $to, $availabilityUser);
+        } catch (Exception $e)  {
```

Comment:
Could you just use multiple catch blocks here, specifying the different types of exceptions each time?

Reply:
done